service/apigateway/internal/awsrestjson/marshal.go

```diff
@@ -19,40 +19,44 @@ type ProtoGetAPIKeyMarshaler struct {
 // This method calls appropriate marshal shape functions as per the input shape and protocol used by the service.
 func (m ProtoGetAPIKeyMarshaler) MarshalOperation(r *aws.Request) {
 	var err error
-	encoder := v2.NewEncoder(r.HTTPRequest)
+	encoder := restV2.NewEncoder(r.HTTPRequest)
 	// adds content-type header
 	encoder.AddHeader("Content-Type").String("application/json")
 
 	err = MarshalGetAPIKeyInputShapeAWSREST(m.Input, encoder)
 	if err != nil {
 		r.Error = err
+		return
 	}
+	encoder.Encode()
 
+	// Todo Instead of passing aws.Request directly to MarshalGetAPIKeyInputShapeAWSJSON;
+	//  we should pass the payload as an argument
 	err = MarshalGetAPIKeyInputShapeAWSJSON(m.Input, r)
 	if err != nil {
 		r.Error = err
+		return
 	}
 }
 
 // MarshalGetAPIKeyInputShapeAWSREST is a stand alone function used to marshal the HTTP bindings a input shape.
 // This method uses the rest encoder utility
-func MarshalGetAPIKeyInputShapeAWSREST(input *types.GetApiKeyInput, encoder *v2.Encoder) error {
+func MarshalGetAPIKeyInputShapeAWSREST(input *types.GetApiKeyInput, encoder *restV2.Encoder) error {
 	// encode using rest encoder utility
 	if err := encoder.SetURI("api_Key").String(*input.ApiKey); err != nil {
-		log.Fatalf("Failed to marshal API KEY to URI using REST encoder:\n \t %v", err.Error())
+		return fmt.Errorf("failed to marshal API KEY to URI using REST encoder:\n \t %v", err.Error())
```

Comment:
Not too important ATM, but these errors should be Serialization errors so we don't have to add the formatting in place here.  Wanted to call this out, but not something we need to address direclty here, but should be thinking about it as we move forward with the prototypes.